server/tapeuse_test.go

```diff
@@ -0,0 +1,205 @@
+//
+
+func upload_string(t *testing.T, verb, route string, s string) string {
+	md5hash := md5.Sum([]byte(s))
+	return upload_stringhash(t, verb, route, s, hex.EncodeToString(md5hash[:]), 200)
+}
+
+func upload_stringhash(t *testing.T, verb, route, s, hash string, statuscode int) string {
+
+	req, err := http.NewRequest(verb, test_server.URL+route, strings.NewReader(s))
+	if err != nil {
+		t.Fatal("Problem creating request", err)
+	}
+	req.Header.Set("X-Upload-Md5", hash)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(route, err)
+		return ""
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != statuscode {
+		t.Errorf("%s: Received status %d, expected %d",
+			route,
+			resp.StatusCode,
+			statuscode)
+		return ""
+	}
+	return resp.Header.Get("Location")
+}
+
+func send_transaction(t *testing.T, route string, commands [][]string, statuscode int) string {
+	content, _ := json.Marshal(commands)
+	return upload_stringhash(t, "POST", route, string(content), "", statuscode)
+}
+
+func get_location(t *testing.T, verb, route string, expstatus int) string {
+	resp := check_route(t, verb, route, expstatus)
+	if resp != nil {
+		resp.Body.Close()
+		return resp.Header.Get("Location")
+	}
+	return ""
+}
+
+func get_body(t *testing.T, verb, route string, expstatus int) string {
+	resp := check_route(t, verb, route, expstatus)
+	if resp != nil {
+		body, err := ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+		if err != nil {
+			t.Fatal(route, err)
+		}
+		return string(body)
+	}
+	return ""
+}
+
+func check_status(t *testing.T, verb, route string, expstatus int) {
+	resp := check_route(t, verb, route, expstatus)
+	if resp != nil {
+		resp.Body.Close()
+	}
+}
+
+func check_route(t *testing.T, verb, route string, expstatus int) *http.Response {
+	req, err := http.NewRequest(verb, test_server.URL+route, nil)
+	if err != nil {
+		t.Fatal("Problem creating request", err)
+	}
+	req.Header.Set("Accept-Encoding", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(route, err)
+		return nil
+	}
+	if resp.StatusCode != expstatus {
+		t.Errorf("%s: Expected status %d and received %d",
+			route,
+			expstatus,
+			resp.StatusCode)
+		resp.Body.Close()
+		return nil
+	}
+	return resp
+}
+
+// waitTrans doesn't return until the given txpath
+// is done processing, either because of success or error,
+// or 100 ms have passed.
+func waitTrans(t *testing.T, txpath string) {
+	for i := 0; i < 10; i++ {
+		time.Sleep(10 * time.Millisecond) // sleep a squinch
+		resp := check_route(t, "GET", txpath, 200)
+		var info struct{ Status transaction.Status }
+		dec := json.NewDecoder(resp.Body)
+		err := dec.Decode(&info)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		if info.Status == transaction.StatusFinished ||
+			info.Status == transaction.StatusError {
+
+			return
+		}
+	}
+	t.Errorf("Timeout waiting for transaction %s", txpath)
+}
+
+var test_server *httptest.Server
+
+func init() {
```

Comment:
It is fine to have two init functions. Don't call this second server `test_server` thought. Maybe call it `itemCacheServer`? since that is the main distinction between this server and the other one spawned...